src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -0,0 +1,105 @@
+import blkid
+import os
+import subprocess
+
+from lxml import etree
+
+from .device_info_base import DeviceInfoBase
+from middlewared.service import private, Service
+
+
+class DeviceService(Service, DeviceInfoBase):
+
+    async def get_serials(self):
+        raise NotImplementedError()
+
+    def get_disks(self):
+        disks = {}
+        lshw_disks = self.retrieve_lshw_disks_data()
+
+        for block_device in filter(
+            lambda b: b.name not in ('sr0',),
+            blkid.list_block_devices()
+        ):
+            disks[block_device.name] = self.get_disk_details(block_device, self.disk_default.copy(), lshw_disks)
+        return disks
+
+    @private
+    def retrieve_lshw_disks_data(self):
+        disks_cp = subprocess.Popen(
+            ['lshw', '-xml', '-class', 'disk'], stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        output, error = disks_cp.communicate()
+        lshw_disks = {}
+        if output:
+            xml = etree.fromstring(output.decode())
+            for child in filter(lambda c: c.get('class') == 'disk', xml.getchildren()):
+                data = {'rotationrate': None}
+                for c in child.getchildren():
+                    if not len(c.getchildren()):
+                        data[c.tag] = c.text
+                    elif c.tag == 'capabilities':
+                        for capability in filter(lambda d: d.text.endswith('rotations per minute'), c.getchildren()):
+                            data['rotationrate'] = capability.get('id')[:-3]
+                lshw_disks[data['logicalname']] = data
+        return lshw_disks
+
+    def get_disk(self, name):
+        disk = self.disk_default.copy()
+        try:
+            block_device = blkid.BlockDevice(os.path.join('/dev', name))
+        except blkid.BlkidException:
+            return disk
+
+        return self.get_disk_details(block_device, disk, self.retrieve_lshw_disks_data())
+
+    @private
+    def get_disk_details(self, block_device, disk, lshw_disks):
+        dev_data = block_device.__getstate__()
+        disk.update({
+            'name': dev_data['name'],
+            'sectorsize': dev_data['io_limits']['logical_sector_size'],
+            'number': ord(dev_data['name'][-1].lower()) - 96,
```

Comment:
We can have more than 26 disks. They are named like that:
```
#1	sda
#2	sdb
...
#26	sdz
#27	sdaa
#28	sdab
...
#52	sdaz
#53	sdba
#54	sdbb
...
```
Also please do not hardcode magic numbers, use `ord('a')`

Reply:
https://github.com/torvalds/linux/blob/master/drivers/scsi/sd.c#L3249 proof